var/spack/repos/builtin/packages/affinity/package.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Affinity(CMakePackage):
```

Comment:
CudaPackage, etc

Reply:
Testing the new version of this on Corona at LLNL, it seems that adding `ROCmPackage` significantly complicates building the package (I need to specify an arch, make sure to `spack external find hsa-rocr-dev`, and manually add an external for `llvm-amdgpu` to my `spack.yaml`. Ultimately all `affinity` does is query what devices are visible to a process. Is the addition of `ROCmPackage` necessary?

Reply:
i guess you can do without then. CudaPackage is still helpful because just #include <cuda_runtime> will cause #errors for unsupported gcc/clang. Unless the affinity package sets some define before including the header.

Reply:
Okay that makes sense, thanks. Once we get better autodetection of the HIP stack, I will add the `ROCmPackage` mixin back in.